packages/odf/components/create-storage-system/external-systems/CreateFlashSystem/FlashSystemConnectionDetails/system-connection-details.tsx

```diff
@@ -72,130 +97,192 @@ export const FlashSystemConnectionDetails: React.FC<
   const onModeSelect = (event, value) => {
     event.preventDefault();
     const volumeMode = volumeMapper[value];
-    setFormState('volmode', volumeMode);
+    setFormState((prev) => ({ ...prev, volmode: volumeMode }));
     setIsOpen(!isOpen);
   };
 
+  const updateFormField = (field: keyof FlashSystemState, value: string) => {
+    setFormState((prev) => ({ ...prev, [field]: value }));
+  };
+
+  // Set up form validation schema
+  const schema = useFlashSystemSchema();
+  const resolver = useYupValidationResolver(schema);
+
+  const { control } = useForm({
+    resolver,
+    mode: 'onChange',
+  });
+
+  const onSubmit = async () => {
+    setIsLoading(true);
+    const payload = createFlashSystemPayload({
+      systemName: 'flash-cluster',
```

Comment:
nits (optional): just to keep behaviour same as previous releases
```suggestion
      systemName: getExternalSubSystemName("IBM FlashSystem Storage", 'flash-storage'),
```

Reply:
also what we were using as StorageClass name so far upto 4.19 ??
I don't think it was `flash-storage` ??